### ISSUE TITLE
unlambda: false positive

### ISSUE THREAD

**Issue by benma** [go-critic/go-critic#888]
```Go
package main

type testEnv struct {
	f func(s string) int
}

var stored func(s string) int

func store(f func(s string) int) {
	stored = f
}

func main() {
	var f func(s string) int
	store(func(s string) int { return f(s) })
	f = func(s string) int { return 42 }
	stored("foo")
}
```

Running `gocritic check -enable unlambda ./...`

```
./snippet.go:23:8: unlambda: replace `func(s string) int { return f(s) }` with `f`
```

This is a false positive, as applying the suggested fix results in different behavior: a panic because `f` is `nil`.

Interestingly, if you remove the `int` return type, the linter does not complain:

```Go
package main

type testEnv struct {
	f func(s string)
}

var stored func(s string)

func store(f func(s string)) {
	stored = f
}

func main() {
	var f func(s string)
	store(func(s string) { f(s) })
	f = func(s string) {}
	stored("foo")
}
```

**Comment by quasilyte** [comment]
Can be fixed by not giving a warning for functions that are called via variable by default.
